dune_client/client.py

```diff
@@ -121,9 +119,13 @@ def refresh(self, query: Query, ping_frequency: int = 5) -> list[DuneRecord]:
         Sleeps `ping_frequency` seconds between each status request.
         """
         job_id = self.execute(query).execution_id
-        while self.get_status(job_id).state != ExecutionState.COMPLETED:
-            log.info(f"waiting for query execution {job_id} to complete")
+        state = self.get_status(job_id).state
+        while state != ExecutionState.COMPLETED:
```

Comment:
here we should wait for the execution to complete, cancel or fail
            # these are the 3 terminal states of an execution

>             terminal_states = (
>                 "QUERY_STATE_COMPLETED",
>                 "QUERY_STATE_FAILED",
>                 "QUERY_STATE_CANCELLED",
>             )

Reply:
Good point! Will add.